droidlet/interpreter/robot/tasks.py

```diff
@@ -620,3 +624,167 @@ def step(self):
             self.agent.mover.explore(self.goal)
             self.finished = self.agent.mover.nav.is_done_exploring().value
 
+class TimedExplore(TrajectorySaverTask):
+    """use slam to explore environemt"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data)
+        self.command_sent = False
+        self.agent = agent
+        self.goal = task_data.get("goal")
+        self.timeout = task_data.get("timeout")
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    @Task.step_wrapper
+    def step(self):
+        super().step()
+        self.interrupted = False
+        self.finished = False
+        if not self.finished and self.timeout > 0:
+            self.agent.mover.explore(self.goal)
+            self.timeout -= 1
+            self.finished = self.agent.mover.nav.is_done_exploring().value
+        else:
+            self.finished = True
+
+
+class Reexplore(Task):
+    """use slam to explore environemt, but also examine detections"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent, task_data)
+        self.steps = ["not_started"] * 5 # S1, C1s, C1l, R1, R2
+        self.task_data = task_data
+        self.target = task_data.get('target')
+        self.spawn_pos = task_data.get('spawn_pos')
+        self.base_pos = task_data.get('base_pos')
+        self.init_logger()
+        self.agent = agent
+        print(f'CuriousExplore task_data {task_data}')
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    def init_logger(self):
```

Comment:
is this some kind of copy-paste for a lot of tasks that you use? can you eliminate / streamline this?

Reply:
yes, planning to. We want to have a state machine kind of thing to execute multi-step tasks like these, and I think logging would also be an inbuilt feature for them. Going to work with @aszlam to streamline this.